pkg/apis/core/validation/shoot.go

```
@@ -1684,3 +1685,41 @@ func validateCoreDNS(coreDNS *core.CoreDNS, fldPath *field.Path) field.ErrorList
 
 	return allErrs
 }
+
+func validateShootOperation(operation string, shoot *core.Shoot, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if operation == "" {
+		return allErrs
+	}
+
+	if !utilfeature.DefaultFeatureGate.Enabled(features.ShootCARotation) {
```

Do we want to return an error if someone sets an operation but the feature gate is set to false?

---

Probably not, it simply wouldn't work. I think that's the "default behaviour" when a feature gate is not turned on.

---

ok